src/Model.elm

```diff
@@ -182,47 +191,140 @@ processFollowEvent relationship flag model =
+
+            UpdateSensitive sensitive ->
+                { model | draft = { draft | sensitive = sensitive } } ! []
+
+            UpdateSpoiler spoiler_text ->
+                { model | draft = { draft | spoiler_text = Just spoiler_text } } ! []
+
+            UpdateVisibility visibility ->
+                { model | draft = { draft | visibility = visibility } } ! []
+
+            UpdateReplyTo status ->
+                { model
+                    | draft =
+                        { draft
+                            | in_reply_to = Just status
+                            , status = Mastodon.Helper.getReplyPrefix currentUser status
+                            , sensitive = Maybe.withDefault False status.sensitive
+                            , spoiler_text =
+                                if status.spoiler_text == "" then
+                                    Nothing
+                                else
+                                    Just status.spoiler_text
+                            , visibility = status.visibility
+                        }
+                }
+                    ! [ Command.focusId "status" ]
+
+            UpdateInputInformation { status, selectionStart } ->
+                let
+                    newModel =
+                        { model
+                            | draft = { draft | status = status }
+                            , autoCursorPosition = selectionStart
+                        }
+
+                    stringToPos =
+                        (String.slice 0 newModel.autoCursorPosition status)
+
+                    atPosition =
+                        (case (String.right 1 stringToPos) of
+                            "@" ->
+                                Just newModel.autoCursorPosition
+
+                            " " ->
+                                Nothing
+
+                            _ ->
+                                model.autoAtPosition
+                        )
+
+                    query =
+                        (case atPosition of
+                            Just position ->
+                                String.slice position (String.length stringToPos) stringToPos
+
+                            Nothing ->
+                                ""
+                        )
+
+                    showMenu =
+                        (not << List.isEmpty <| (acceptableAccounts query model.autoAccounts))
+                in
+                    { newModel
+                        | autoAtPosition = atPosition
+                        , autoQuery = query
+                        , showAutoMenu = showMenu
+                    }
+                        ! [ Command.searchAccounts model.client query model.autoMaxResults False ]
+
+            SelectAccount id ->
+                let
+                    account =
+                        Debug.log "[Account] " <| ((List.filter (\account -> (toString account.id) == id) model.autoAccounts) |> List.head)
```

Comment:
Nit: Don't forget to remove this before it lands. Also this line would fit better spread on multiple lines.